src/DynamoCoreWpf/Views/Core/WorkspaceView.xaml.cs

```diff
@@ -85,6 +87,11 @@ public WorkspaceView()
 
         void OnWorkspaceViewLoaded(object sender, RoutedEventArgs e)
         {
+            // The synchronization context is handing down to the view
+            // model so that operations which need to be executed on the 
+            // UI thread can do so.
+            ViewModel.Model.Context = SynchronizationContext.Current;
```

Comment:
How can you be sure `SynchronizationContext.Current` is what you expect?  What if it's never set?  It's not a good idea to introduce public immutable state like this as anyone can dip their hands in the pot, so to speak.